app/interactors/tasks/update.rb

```diff
@@ -1,7 +1,13 @@
 module Tasks
   class Update
     include Interactor::Organizer
+    
+    delegate :task, :project, to: :context
 
     organize Tasks::Save
+
+    after do
+      TaskMailer.task_updated(project, task).deliver_later
+    end
```

Comment:
And remove project argument in `tasks_controller.rb`

```
delegate :project, to: :task

==

def project
  @project ||= task.project
end
```

```suggestion
    delegate :task, to: :context
    delegate :project, to: :task

    organize Tasks::Save

    after do
      TaskMailer.task_updated(project, task).deliver_later
    end
```